src/solvers.jl

```diff
@@ -573,17 +573,21 @@ function vartypes_without_fixed(m::Model)
     return colCats
 end
 
-function collect_expr!(m, tmprow, terms::AffExpr)
+# Collect the terms of the expression `terms` for which the model of the variable is `m`
+# into `tmprow`. If the variable of one of the terms does not belong to the model `m` and
+# `ignore_not_owned` is `false` then an error is thrown.
+function collect_expr!(m::Model, tmprow::IndexedVector, terms::AffExpr, ignore_not_owned::Bool=false)
     empty!(tmprow)
     assert_isfinite(terms)
     coeffs = terms.coeffs
     vars = terms.vars
     # collect duplicates
     for ind in 1:length(coeffs)
-        if vars[ind].m !== m
+        if vars[ind].m === m
+            addelt!(tmprow, vars[ind].col, coeffs[ind])
+        elseif checkowned
```

Comment:
Looking at codecov, it seems it wasn't covered by the tests. I have added a test in c715a24